### ISSUE TITLE
Random Small Delay in Hot Reload During React Web Development

### ISSUE THREAD

**Issue by fanesz** [zen-browser/desktop#2301]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

This issue has been bothering me for a while when doing web development using Zen Browser.
It can happen randomly, but the way I can consistently reproduce the bug is shown in this video:

[![video]()](https://github.com/user-attachments/assets/36930fc8-afde-4be4-934e-1bf7f0efa5fd)

The delay only happens on Zen Browser; there is no delay on Firefox, Edge, or Chrome.

Steps:
1. Initial react project (npm/bun create vite@latest) & run dev
2. open it using zen browser
3. do any changes (component/style changes)
4. got a delay on hot reload feature

Framework/lib I use:
- vite 5.4.9
- react 18.3.1
- typescript 5.3.1

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.12 (Firefox 131.0.3) (64-bit)

### What platform are you seeing the problem on?

Windows

**Comment by salamijason** [comment]
Hey are you still getting the delay on the latest ver? (1.0.1-a.14 as of Oct 29)

**Comment by fanesz** [comment]
Sorry for the late response, yep I just tested it, and still got the delay
@salamijason 
1.0.1-a.14 (Firefox 131.0.3) (64-bit)

**Comment by salamijason** [comment]
have been notified that this might be because of input delay caused from box shadows.

**Comment by fanesz** [comment]
>     * Please let us know if this issue is still relevant to the latest version of the desktop repository by commenting here.

yes, I can confirm that the bug still happen with latest version of zen (1.0.1-a.22 (Firefox 132.0.1) (64-bit))

**Comment by mr-cheffy** [comment]
I think this may just be due to a performance loss?

**Comment by jakobrosenberg** [comment]
I believe this might be related.
https://github.com/vitejs/vite/issues/20151
